client/packages/system/src/Stock/ListView/ListView.tsx

```diff
@@ -160,13 +158,16 @@ const StockListComponent: FC = () => {
       sortBy,
       onChangeSortBy: updateSortQuery,
     },
-    [sortBy, pluginColumns]
+    [sortBy, plugins.stockColumn?.columns.length || 0]
   );
 
   return (
     <>
       <Toolbar filter={filter} />
       <AppBarButtons />
+      {plugins.stockColumn?.StateLoader?.map((StateLoader, index) => (
```

Comment:
There is a state loader for columns, which allows to get all data for all columns in one go